Library/Homebrew/cli/named_args.rb

```diff
@@ -174,10 +174,12 @@ def resolve_keg(name)
         end
       end
 
-      def cask_exists_with_ref(ref)
-        Cask::CaskLoader.load ref
+      def puts_treatment_if_cask_exists(ref, treated_as)
+        cask = Cask::CaskLoader.load ref
+
+        puts "Treating #{ref} as a #{treated_as}. For the cask, use #{cask.tap.name}/#{cask.token}"
       rescue Cask::CaskUnavailableError
-        false
+        # No ref conflict with a cask, do nothing
```

Comment:
Why is `Cask::CaskUnavailableError` not a problem?

Reply:
This method prints a warning if there is a cask with the same name as a formula/keg. If there are no such casks, then there's nothing to do

Reply:
oh! that's because you combined `puts` with `if cask_exists_with_ref name` into `warn_if_cask_conflicts`. makes sense